pkg/provider/provider_acceptance_test.go

```diff
@@ -178,13 +178,49 @@ func TestAcc_Provider_useNonExistentDefaultParams(t *testing.T) {
 				Config:      providerConfigWithWarehouse(testprofiles.Default, nonExisting),
 				ExpectError: regexp.MustCompile("The requested warehouse does not exist or not authorized."),
 			},
+			// check that using a non-existing warehouse with disabled verification succeeds
 			{
 				Config: providerConfigWithWarehouseAndDisabledValidation(testprofiles.Default, nonExisting),
 			},
 		},
 	})
 }
 
+// prove we can use tri-value booleans, similarly to the ones in resources
+func TestAcc_Provider_triValueBoolean(t *testing.T) {
+	t.Setenv(string(testenvs.ConfigureClientOnce), "")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			acc.TestAccPreCheck(t)
+			testenvs.AssertEnvNotSet(t, snowflakeenvs.User)
+			testenvs.AssertEnvNotSet(t, snowflakeenvs.Password)
+		},
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		Steps: []resource.TestStep{
+			{
+				ExternalProviders: map[string]resource.ExternalProvider{
```

Comment:
nit: you had in the doc a proposal to extract short func to setup such tests, so maybe let's add this method and use it in the new tests (even without using it in all the places)

Reply:
I can do this in the next pr.